mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1485,7 +1485,7 @@ def _evaluate(
 
             text_metrics = [toxicity, perplexity, flesch_kincaid_grade_level, ari_grade_level]
 
-            if self.model_type not in _ModelType.values():
+            if self.model_type not in _ModelType.values() and self.model_type is not None:
```

Comment:
Can we remove this if check to use `can_evaluate` so the logic for checking this is only there in 1 place?